src/flash-media-source.js

```diff
@@ -36,6 +37,12 @@ export default class FlashMediaSource extends videojs.EventTarget {
         }
       });
 
+      if (this.tech_.hls) {
+        this.tech_.hls.on('dispose', () => {
```

Comment:
Techs do have a dispose but they are only dispose when switching techs (html->flash) but sourceHandlers have a dispose too and they trigger it whenever the source changes (their dispose gets triggered via HLSHandler's superclass in `Component#dispose`).

So while I agree that this line is kind of cludgy, it would be necessary since video.js will properly clean up remote text tracks whenever the source changes. This is just insurance that they get cleaned up even if using an older version of video.js.